code/modules/mob/living/carbon/xenomorph/abilities/predalien/predalien_powers.dm

```diff
@@ -42,116 +39,183 @@
+				if(!isxeno_human(human) || xeno.can_not_harm(human))
+					continue
+
+				if (human.stat == DEAD)
+					continue
+
+				if(!check_clear_path_to_target(xeno, human))
+					continue
+
+				if (range > 1)
+					xeno.visible_message(SPAN_XENOHIGHDANGER("[xeno] rips open the guts of [human]!"), SPAN_XENOHIGHDANGER("We rip open the guts of [human]!"))
+					human.spawn_gibs()
+					playsound(get_turf(human), 'sound/effects/gibbed.ogg', 30, 1)
+					human.apply_effect(get_xeno_stun_duration(human, 1), WEAKEN)
+				else
+					xeno.visible_message(SPAN_XENODANGER("[xeno] claws [human]!"), SPAN_XENODANGER("We claw [human]!"))
+					playsound(get_turf(human), "alien_claw_flesh", 30, 1)
+
+				human.apply_armoured_damage(get_xeno_damage_slash(human, aoe_damage + aoe_scale * behavior.kills), ARMOR_MELEE, BRUTE, "chest", 20)
+			playsound(owner, 'sound/voice/predalien_growl.ogg', 75, 0, status = 0)
+
+		var/list/mobs_in_range = oviewers(xeno)
+		for(var/mob/mob as anything in mobs_in_range)
+			if(mob.stat == DEAD || HAS_TRAIT(mob, TRAIT_NESTED))
+				continue
+
+
+		REMOVE_TRAIT(xeno, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("Eviscerate"))
+		xeno.anchored = FALSE
+		apply_cooldown()
+		return
+
+	var/mob/living/carbon/xenomorph/x = owner
 
 	if (!action_cooldown_check())
 		return
 
 	if (!xeno.check_state())
 		return
 
-	var/datum/behavior_delegate/predalien_base/behavior = xeno.behavior_delegate
-	if(!istype(behavior))
+	if (!isxeno_human(target) || x.can_not_harm(target))
+		to_chat(xeno, SPAN_XENOWARNING("We must target a hostile!"))
 		return
 
-	if(!check_plasma_owner())
+	if (get_dist_sqrd(target, xeno) > 2)
+		to_chat(xeno, SPAN_XENOWARNING("[target] is too far away!"))
 		return
 
-	if(!do_after(xeno, activation_delay, INTERRUPT_ALL, BUSY_ICON_GENERIC))
-		to_chat(xeno, "Keep still whilst trying to smash into the ground")
-
-		var/real_cooldown = xeno_cooldown
+	var/mob/living/carbon/carbon = target
 
-		xeno_cooldown = 3 SECONDS
-		apply_cooldown()
-		xeno_cooldown = real_cooldown
+	if (carbon.stat == DEAD)
+		to_chat(xeno, SPAN_XENOWARNING("[carbon] is dead, why would we want to touch them?"))
 		return
 
-	if(!check_and_use_plasma_owner())
-		return
+	if(targetting_type == SINGLETARGETGUT)
+		var/datum/behavior_delegate/predalien_base/predalienbehavior = xeno.behavior_delegate
+		if(!istype(predalienbehavior))
+			return
 
-	playsound(xeno.loc, pick(smash_sounds), 50, 0, status = 0)
-	xeno.visible_message(SPAN_XENOHIGHDANGER("[xeno] smashes into the ground!"))
+		if (!check_and_use_plasma_owner())
+			return
 
-	xeno.create_stomp()
+		ADD_TRAIT(carbon, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("Devastate"))
 
-	for(var/mob/living/carbon/carbon in oview(round(behavior.kills * 0.5 + 2), xeno))
-		if(!xeno.can_not_harm(carbon) && carbon.stat != DEAD)
-			ADD_TRAIT(carbon, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("Smash"))
+		if (ishuman(carbon))
+			var/mob/living/carbon/human/human = carbon
+			human.update_xeno_hostile_hud()
```

Comment:
Have a think about if this is necessary, I am not sure before I pull this and test.